lua/treesitter-context.lua

```diff
@@ -392,31 +392,25 @@ end
 
 --- @generic F: function
 --- @param fn F
+--- @param ms? number
 --- @return F
-local function throttle_fn(fn)
-  local recalc_after_cooldown = false
-  local cooling_down = false
-  local function wrapped()
-    if cooling_down then
-      recalc_after_cooldown = true
-    else
-      local start = vim.loop.hrtime()
-      fn()
-      local elapsed_ms = math.floor((vim.loop.hrtime() - start) / 1e6)
-      -- If this took < 2ms, we don't need a cooldown period. This prevents the context floats from flickering
-      if elapsed_ms > 2 then
-        cooling_down = true
-        vim.defer_fn(function()
-          cooling_down = false
-          if recalc_after_cooldown then
-            recalc_after_cooldown = false
-            wrapped()
-          end
-        end, 20)
-      end
+local function debounce(fn, ms)
```

Comment:
This is no longer a debounce right as fn will be called at most every 100ms? A debounce could delay indefinitely.

Reply:
Still a debounce, but will additionally trigger on the leading edge

Reply:
So this can delay indefinitely? Shouldn't it be called at least every 20ms?

Reply:
nvm let's just go with this.